elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/impl/ElasticsearchBackendQueueProcessor.java

```diff
@@ -24,10 +35,12 @@
 public class ElasticsearchBackendQueueProcessor implements BackendQueueProcessor {
 
 	private ElasticsearchIndexManager indexManager;
+	private ExtendedSearchIntegrator searchIntegrator;
 
 	@Override
 	public void initialize(Properties props, WorkerBuildContext context, IndexManager indexManager) {
 		this.indexManager = (ElasticsearchIndexManager) indexManager;
+		this.searchIntegrator = context.getUninitializedSearchIntegrator();
```

Comment:
Sorry to be late to the party! I have a question about this one: is there a reason why you don't use the searchIntergrator from the indexManager as it's done below?

If so would it be an acceptable way to initialize the ElasticsearchIndexWorkVisitor in the constructor once and for all?